# In this script I investigate the files generated by Qiime2 and how our dataset look like in regards to number of samples.
# Libraries ----
library(tidyverse)
library(gplots)

# Palette ----
load("../../3rd_LesionRNAseq_Dataset/RStudio_outputs/data/Dark24")

# Import metadata ----
metadata <- read_delim("~/Box Sync/human_lesion_microbiome/Camila_MasterMetadata_DONOTedit/Camila16SseqProcessing/MiSeqV1V3_38_humanLesion_metadata_removedLowReadSamples.tsv", 
           "\t", escape_double = FALSE, col_types = cols(SubjectID = col_character(), 
                                                         Swab_site = col_factor(levels = c("environmental", 
                                                                                           "contralateral", "lesion"))), trim_ws = TRUE)
colnames(metadata)[6] <- "LTCP_patient_ID"

# number of features from table.qzv ----
tableqzv <- read_delim("table.txt", "\t", escape_double = FALSE, 
                    trim_ws = TRUE)
colnames(tableqzv)[1] <- colnames(metadata)[1]

# Merging metadata with Feature counts
metadata <- left_join(metadata, tableqzv, by=colnames(tableqzv)[1])

# And plotting:
metadata %>%
ggplot(.,
       aes(x=Swab_site, y=metadata$`Feature Count`, color=Swab_site)) +
  geom_violin() +
  geom_jitter(position=position_jitter(0.1), size=1,alpha=.2) +
  theme_classic() + scale_color_manual(values = Dark24) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        axis.text.x = element_text(size = 11, angle = 45, hjust = 1), 
        axis.text.y = element_text(size = 11), axis.title = element_text(size = 11),
        legend.position="none") +
  xlab("") + ylab(paste("Number of counts post-QC"))

# The data we have in this 16S-seq ----
#'%ni%' <- Negate('%in%')
metadata_mod <- metadata %>%
  filter(Swab_site %in% "lesion") %>%
  select(LTCP_patient_ID,Swab_site,Time_point) %>%
  mutate(Time_point1 = case_when(str_detect(Time_point, '^day') ~ 1)) %>%
  pivot_wider(names_from = Time_point, values_from = Time_point1) %>%
  filter(day0 == 1) %>%
  arrange(day0, day30, day60, day90, day120, day150, day180, day210, day240,day270) %>%
  column_to_rownames("LTCP_patient_ID") %>%
  select(day0, day30, day60, day90, day120, day150, day180, day210, day240,day270) %>%
  replace(is.na(.), 0)
save(metadata_mod, file="metadata_mod")


# matrix:
colorheat <- colorRampPalette(colors=c("white",Dark24[1]))(2)
heatmap.2(as.matrix(metadata_mod), 
          dendrogram = "none",
          Colv=F, Rowv = F,
          col=colorheat, 
          scale='row',
          density.info="none", trace="none",  
          cexCol=1, cexRow = .7,
          colsep=1:nrow(metadata_mod2),
          rowsep=1:nrow(metadata_mod2),
          sepcolor = "black")

  
